src/frontend/src/catalog/root_catalog.rs

```diff
@@ -968,12 +968,10 @@ impl Catalog {
             Err(CatalogError::duplicated("sink", relation_name.to_owned()))
         } else if schema.get_view_by_name(relation_name).is_some() {
             Err(CatalogError::duplicated("view", relation_name.to_owned()))
-        } else if let Some(subscription) = schema.get_subscription_by_name(relation_name) {
-            let is_not_created = subscription.subscription_state != SubscriptionState::Created;
-            Err(CatalogError::Duplicated(
+        } else if schema.get_subscription_by_name(relation_name).is_some() {
```

Comment:
Here if a subscription with the same name is being creating (with `subscription.subscription_state != SubscriptionState::Created`), we will return an error message without the `under creation`, and then some of our tests will treat the subscription as created. However, the subscription is not actually created, and later operation depending on the subscription will fail. We've observed related failure in our CI.

For simplicity we may just keep the logic here unchanged, or we will have to handle the `if_not_exists` in the `CreateSubscription` ddl service in meta.